lib/datadog/core/telemetry/logging.rb

```diff
@@ -5,13 +5,48 @@
 module Datadog
   module Core
     module Telemetry
-      # Logging interface for sending telemetry logs.
+      # === INTRENAL USAGE ONLY ===
+      #
+      # Logging interface for sending telemetry logs... so we can fix them.
+      #
+      # For developer using this module:
+      # - MUST NOT provide any sensitive information (PII)
+      # - SHOULD reduce the data cardinality for batching/aggregation
+      #
+      # Before using it, ask yourself:
+      # - Do we need to know about this (ie. internal error or client error)?
+      # - How severe/critical is this error? (ie. error, warning, fatal)
+      # - What information needed to make it actionable?
       #
-      # Reporting internal error so that we can fix them.
-      # IMPORTANT: Make sure to not log any sensitive information.
       module Logging
         extend self
 
+        # Extract datadog stack trace from the exception
+        module DatadogStackTrace
+          # Typically, `lib` is under `#{gem_name}-#{version}/`
+          # but not the case when providing a bundler custom path in `Gemfile`
+          REGEX = %r{/lib/datadog/}.freeze
+
+          def self.from(exception)
+            backtrace = exception.backtrace
+
+            return unless backtrace
+            return if backtrace.empty?
+
+            stack_trace = +''
+            backtrace.each do |line|
+              stack_trace << if line.match?(REGEX)
+                               # Removing host related information
+                               line.sub(/^.*?(#{REGEX})/o, '\1') << ','
+                             else
+                               'REDACTED,'
+                             end
+            end
```

Comment:
So while this approach works, I'm not the biggest fan as:

* It matches on every `lib/datadog`, not just ours (e.g. the customer may have a datadog folder where they put some datadog setup they use)
* There's a bit of overhead in doing regexp matching and replacement

Have you considered prefix matching? E.g. instead of `REGEX = ...` you can get the root of the gem folder by starting from the current file `GEM_ROOT = Pathname.new("#{__dir__}/../../").cleanpath.to_s`

Then, you can match using `.start_with?` and use string slicing to remove the prefix.